edx_exams/apps/api/v1/views.py

```diff
@@ -1 +1,164 @@
+                # add to visited list so that we know to not mark this exam as inactive later
+                visited_exams.append(exam['content_id'])
+
+                if exam_object:
+                    if exam['exam_type'] != exam_object.exam_type:
+                        # if the exam type has changed, mark existing exam as inactive and then create
+                        # a new exam
+                        exam_object.is_active = False
+                        exam_object.save()
+
+                        current = Exam.objects.create(
+                            resource_id=str(uuid.uuid4()),
+                            course_id=exam_object.course_id,
+                            provider=exam_object.provider,
+                            content_id=exam_object.content_id,
+                            exam_name=exam['exam_name'],
+                            exam_type=exam['exam_type'],
+                            time_limit_mins=exam['time_limit_mins'],
+                            due_date=exam['due_date'],
+                            hide_after_due=exam['hide_after_due'],
+                            is_active=exam['is_active']
+                        )
+                        log.info(
+                            "Marked exam=%(previous_exam_id)s as inactive, created new exam=%(current_exam_id)s",
+                            {
+                                'previous_exam_id': exam_object.id,
+                                'current_exam_id': current.id,
+                            }
+                        )
+                    else:
+                        # if exam type has not changed, update the fields of the existing exam
+                        exam_object.exam_name = exam['exam_name']
+                        exam_object.time_limit_mins = exam['time_limit_mins']
+                        exam_object.due_date = exam['due_date']
+                        exam_object.hide_after_due = exam['hide_after_due']
+                        exam_object.is_active = exam['is_active']
+                        exam_object.save()
+                        log.info(
+                            "Updated existing exam=%(current_exam_id)s",
+                            {
+                                'current_exam_id': exam_object.id,
+                            }
+                        )
+
+                else:
+                    # if exam does not already exist, create a new one
+
+                    # check that configuration exists
+                    config = CourseExamConfiguration.objects.filter(course_id=exam['course_id'])
+
+                    if len(config) > 0:
+                        provider = config[0].provider
+
+                    else:
+                        # if configuration doesn't already request, return 400
+                        error_msg = "Cannot update exam. No exam configuration set up for course"
+                        return Response(error_msg, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
Decided to return 400 if there is no config. If we want, we could create a default config, but not sure what those values would be, i.e. who is our default provider?

Reply:
There's definitely a user flow thing here where we have to decide if we want course teams to be able to publish timed/proctored exams prior to explicitly choosing a provider. Which gets me thinking what about non-proctored exams these should probably be set to null anyway 🤔

Reply:
The provider is now set to None if the exam is either timed, or if it is set to proctored but there is no config set up for the course. 

Thoughts on allowing None for the second case? Do we want a placeholder config (what would that look like)?

Reply:
None sounds fine for the moment, we'll probably have to revisit this later.